docs/examples/bayesian-example-NGC6440E.py

```diff
@@ -0,0 +1,203 @@
+# %% [markdown]
+# # PINT Bayesian Interface Examples
+
+# %%
+from pint.models import get_model, get_model_and_toas
+from pint.bayesian import BayesianTiming
+from pint.config import examplefile
+from pint.models.priors import Prior
+from scipy.stats import uniform
+
+# %%
+import numpy as np
+import emcee
+import nestle
+import corner
+import io
+import matplotlib.pyplot as plt
+
+# %%
+# Read the par and tim files
+parfile = examplefile("NGC6440E.par.good")
+timfile = examplefile("NGC6440E.tim")
+model, toas = get_model_and_toas(parfile, timfile)
+
+# %%
+# This is optional, but the likelihood function behaves better if
+# we have the pulse numbers. Make sure that your timing solution is
+# phase connected before doing this.
+toas.compute_pulse_numbers(model)
+
+# %%
+# Now set the priors.
+# I am cheating here by setting the priors around the maximum likelihood estimates.
+# This is a bad idea for real datasets and can bias the estimates. I am doing this
+# here just to make everything finish faster. In the real world, these priors should
+# be informed by, e.g. previous (independent) timing solutions, pulsar search results,
+# VLBI localization etc. Note that unbounded uniform priors don't work here.
+for par in model.free_params:
+    param = getattr(model, par)
+    param_min = float(param.value - 10 * param.uncertainty_value)
+    param_span = float(20 * param.uncertainty_value)
+    param.prior = Prior(uniform(param_min, param_span))
```

Comment:
So you are setting an attribute that persist in the model object?   Can you also include in the example how you set the priors with the separate dictionary?

Reply:
> So you are setting an attribute that persist in the model object?

The model attribute in BayesianTiming is supposed to be a clone of the input model. So line 39 should have been `param = getattr(self.model, par)`. I have fixed this.

Reply:
I have included the `prior_info` usage in `bayesian-example-NGC6440E.py` and a new test for this in `test_bayesian.py`.